analyzers/src/SonarAnalyzer.CSharp/Extensions/ExpressionSyntaxExtensions.Roslyn.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.CodeAnalysis.CSharp.Extensions;
+
+// Copied from Roslyn
+// https://github.com/dotnet/roslyn/blob/575bc42589145ba18b4f1cc2267d02695f861d8f/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ExpressionSyntaxExtensions.cs
+[ExcludeFromCodeCoverage]
+public static class ExpressionSyntaxExtensions
+{
+    // Copy of
+    // "https://github.com/dotnet/roslyn/blob/575bc42589145ba18b4f1cc2267d02695f861d8f/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ExpressionSyntaxExtensions.cs#L319"
+    public static bool IsWrittenTo(
```

Comment:
In the VB.NET version below, I propose to add a documentation comment to explain what the method does. Maybe the best place to add such documentation would be the language or syntax facade, where the `IsWrittenTo` is wrapped.

Reply:
I see you went for inheriting the doc from the copied extension method to the facade. Good solution as well.
From the perspective of the user of the method in the facade, it shouldn't change much (provided he is in the context of the IDE, otherwise he has to follow the link in the doc).